flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/TwoInputStreamTaskTest.java

```diff
@@ -473,7 +473,7 @@ public void testOperatorMetricReuse() throws Exception {
                     @Override
                     public OperatorMetricGroup getOrAddOperator(
                             OperatorID operatorID, String name) {
-                        return new OperatorMetricGroup(
+                        return OperatorMetricGroup.createOperatorMetricGroup(
```

Comment:
Let's do it like this instead:

```
final TaskMetricGroup taskMetricGroup =
        MetricUtils.createTaskManagerMetricGroup(
                NoOpMetricRegistry.INSTANCE,
                "host",
                ResourceID.generate())
        .addTaskForJob(
                new JobID(),
                "jobname",
                new JobVertexID(),
                new ExecutionAttemptID(),
                "task",
                0,
                0))
```

The current is exactly the kind of hacks that I want to get rid of.

This way we also won't need `createOperatorMetricGroup`.

Reply:
> Let's do it like this instead:
> 
> ```
> final TaskMetricGroup taskMetricGroup =
>         MetricUtils.createTaskManagerMetricGroup(
>                 NoOpMetricRegistry.INSTANCE,
>                 "host",
>                 ResourceID.generate())
>         .addTaskForJob(
>                 new JobID(),
>                 "jobname",
>                 new JobVertexID(),
>                 new ExecutionAttemptID(),
>                 "task",
>                 0,
>                 0))
> ```
> 
> The current is exactly the kind of hacks that I want to get rid of.
> 
> This way we also won't need `createOperatorMetricGroup`.

Got it, I will update here, thanks!

Reply:
hmm could we do something like this?

```
        TaskMetricGroup taskMetricGroup =
                new UnregisteredMetricGroups.UnregisteredTaskMetricGroup() {
                    @Override
                    public OperatorMetricGroup getOrAddOperator(
                            OperatorID operatorID, String name) {
                        OperatorMetricGroup operatorMetricGroup = super.getOrAddOperator(operatorID, name);
                        operatorMetrics.put(name, operatorMetricGroup);
                        return operatorMetricGroup;
                    }
                };
```

Reply:
> hmm could we do something like this?
> 
> ```
>         TaskMetricGroup taskMetricGroup =
>                 new UnregisteredMetricGroups.UnregisteredTaskMetricGroup() {
>                     @Override
>                     public OperatorMetricGroup getOrAddOperator(
>                             OperatorID operatorID, String name) {
>                         OperatorMetricGroup operatorMetricGroup = super.getOrAddOperator(operatorID, name);
>                         operatorMetrics.put(name, operatorMetricGroup);
>                         return operatorMetricGroup;
>                     }
>                 };
> ```

Oh good point! Sorry I forgot `UnregisteredTaskMetricGroup` extends `TaskMetricGroup`, thanks!